internal/app/app_suite_test.go

```diff
@@ -32,7 +32,7 @@ import (
 
 const (
 	pollingInterval      = 50 * time.Millisecond
-	eventuallyTimeout    = 5 * time.Second
+	eventuallyTimeout    = 10 * time.Second
```

Comment:
Any reason why we should increase the timeout here?

Reply:
There is one eventually block in `core_test.go` which throws timeout error with 5 seconds, so I thought to increase the timeout, also this error occurs sporadically with github runner. It does not give any timeout error while running tests in local machine.

Reply:
Reverted the `eveltuallyTimeout` to 3s. Let's handle this timeout error in another PR and follow up in another issue.